### PR TITLE
refactor #1620: migrated notification screen to compose

### LINKED ISSUE #1620: refactor: migrate notification screen to compose

### PR DESCRIPTION
**Pull request by PratyushSingh07**
## Issue Fix
Fixes #1620 

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifospay/src/main/java/org/mifospay/notification/presenter/NotificationViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                override fun onError(message: String) {
+                    _notificationUiState.value = NotificationUiState.Error(message)
+                }
+            })
+    }
+}
+
+sealed interface NotificationUiState {
+    data object Loading : NotificationUiState
+    data object Empty : NotificationUiState
```
Comment (AvneetSingh2001) [inline mifospay/src/main/java/org/mifospay/notification/presenter/NotificationViewModel.kt]: you can remove this empty state

File: mifospay/src/main/java/org/mifospay/notification/presenter/NotificationViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    val notificationUiState: StateFlow<NotificationUiState> = _notificationUiState
+
+    private val _isRefreshing = MutableStateFlow(false)
+    val isRefreshing: StateFlow<Boolean> get() = _isRefreshing.asStateFlow()
+
+    fun refresh() {
+        viewModelScope.launch {
+            _isRefreshing.emit(true)
+            delay(1000)
+            _isRefreshing.emit(false)
```
Comment (AvneetSingh2001) [inline mifospay/src/main/java/org/mifospay/notification/presenter/NotificationViewModel.kt:35]: move this to fetchNotification()

File: mifospay/src/main/java/org/mifospay/notification/ui/NotificationActivity.kt [hunk 3]
```diff
@@ -26,77 +28,21 @@ import javax.inject.Inject
...
+//        mNotificationPresenter?.fetchNotifications()
     }
 
-    override fun setPresenter(presenter: NotificationContract.NotificationPresenter?) {
-        mNotificationPresenter = presenter
-    }
+//    private fun setupSwipeRefreshLayout() {
+//        setSwipeRefreshEnabled(true)
+//        swipeRefreshLayout?.setOnRefreshListener { mNotificationPresenter?.fetchNotifications() }
+//    }
```
Comment (AvneetSingh2001) [inline mifospay/src/main/java/org/mifospay/notification/ui/NotificationActivity.kt]: remove commented code

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -65,6 +66,7 @@ androidxBrowser = "1.8.0"
 playServicesCodeScanner = "16.1.0"
 
 [libraries]
+accompanist-swiperefresh = { module = "com.google.accompanist:accompanist-swiperefresh", version.ref = "accompanistSwiperefreshVersion" }
```
Comment (therajanmaurya) [inline gradle/libs.versions.toml]: It is depreciated https://google.github.io/accompanist/swiperefresh/ library

Comment (therajanmaurya) [reply gradle/libs.versions.toml]: This library is deprecated, with official pull refresh support in [androidx.compose.material.pullrefresh](https://developer.android.com/reference/kotlin/androidx/compose/material/pullrefresh/package-summary). The migration guide and original documentation is below.

File: mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                NotificationUiState.Loading -> {
+                    MfLoadingWheel(
+                        contentDesc = stringResource(R.string.loading),
+                        backgroundColor = Color.White
+                    )
+                }
+
+                is NotificationUiState.Success -> {
+                    if (uiState.notificationList.isEmpty()) {
+                        Column(
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt]: We have UI to show the empty data 
``` 
EmptyContentScreen(
                        modifier = Modifier,
                        title = stringResource(id = R.string.error_oops),
                        subTitle = stringResource(id = R.string.unexpected_error_subtitle),
                        iconTint = Color.Black,
                        iconImageVector = Icons.Rounded.Info
)
```

File: mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+        shape = RoundedCornerShape(12.dp)
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(24.dp)
+        ) {
+            Text(
+                text = title,
+                style = styleMedium16sp,
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt]: Use material theme `MaterialTheme.typography.bodySmall` 

`bodySmall` will be according to test styling. check in designsystem/MifosTypography which is perfect for the text

File: mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+                text = timestamp, style = historyItemTextStyle,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(bottom = 8.dp)
+            )
+        }
+    }
+}
+
+@Preview(showSystemUi = true, showBackground = true)
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt]: Talk to @AvneetSingh2001 about the generating preview for different states and make change

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: gradle/libs.versions.toml (hunk 4 above)
Comment (therajanmaurya) [reply gradle/libs.versions.toml]: Remove this

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

File: gradle/libs.versions.toml [hunk 8]
```diff
@@ -1,4 +1,5 @@
 [versions]
+accompanistSwiperefreshVersion = "0.27.0"
```
Comment (therajanmaurya) [inline gradle/libs.versions.toml]: Remove this

File: mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+                        contentDesc = stringResource(R.string.loading),
+                        backgroundColor = Color.White
+                    )
+                }
+
+                is NotificationUiState.Success -> {
+                    if (uiState.notificationList.isEmpty()) {
+                        EmptyContentScreen(
+                            modifier = Modifier,
+                            title = stringResource(id = R.string.error_oops),
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt]: Text is wrong, It should be something like 
title - Nothing to Notify
subtitle - There is nothing to show

File: mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+        PullRefreshIndicator(
+            refreshing = isRefreshing,
+            state = pullRefreshState,
+            modifier = Modifier.align(Alignment.TopCenter)
+        )
+    }
+}
+
+@Composable
+fun NotificationList(title: String, body: String, timestamp: String) {
```
Comment (therajanmaurya) [inline mifospay/src/main/java/org/mifospay/notification/ui/NotificationScreen.kt]: change function name to `NotificationListItem`

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Review by therajanmaurya — APPROVED** [review]

**Pull request merged**